src/parser.c

```diff
@@ -70,40 +76,42 @@ static void ParseOperator(const TokenType operation) {
   }
 }
 
-static void ParseNumericExpression() {
+// NOLINTBEGIN(misc-no-recursion)
+static void ParseNumericExpression(const Precedence precedence) {
   ParseNumber();
 
   ConsumeNextToken();
 
-  if (kTokenRightParenthesis == token.type) {
-    source_code--;
-
-    return;
-  }
-
-  while (kTokenPlus == token.type || kTokenMinus == token.type ||
-         kTokenStar == token.type || kTokenGreaterThan == token.type ||
-         kTokenSlash == token.type || kTokenLessThan == token.type ||
-         kTokenGreaterOrEquals == token.type ||
-         kTokenLessOrEquals == token.type) {
+  while ((kTokenPlus == token.type || kTokenMinus == token.type ||
+          kTokenStar == token.type || kTokenGreaterThan == token.type ||
+          kTokenSlash == token.type || kTokenLessThan == token.type ||
+          kTokenGreaterOrEquals == token.type ||
+          kTokenLessOrEquals == token.type || kTokenEquals == token.type ||
+          kTokenNotEquals == token.type) &&
+         precedence <= token.precedence) {
     const TokenType kOperation = token.type;
+    const unsigned char kNextPrecedence = token.precedence + 1;
```

Comment:
@WannabeMO `token.precedence` is an enum value, so it is most likely an `int`, not `unsigned char`, but double check if that's true.

```suggestion
    const int kNextPrecedence = token.precedence + 1;
```

Reply:
Using unsigned char is another way to use integers in C, while having a size of 1 byte instead of the usual 4 bytes. (I believe, I'm no too sure).